ts/src/transports/fetch.ts

```diff
@@ -3,10 +3,21 @@ import {Transport, TransportOptions} from "./Transport";
 import {debug} from "../debug";
 import detach from "../detach";
 
+interface FetchOptions {
+  credentials?: RequestCredentials;
+}
+
+/* fetchRequestWithOptions provides a way to construct a Fetch transport with custom FetchOptions */
+export function fetchRequestWithOptions(fetchOptions: FetchOptions = {}): (options: TransportOptions) => Transport {
+  return function(options: TransportOptions) {
+    return fetchRequest(options, fetchOptions);
+  }
+}
+
 /* fetchRequest uses Fetch (with ReadableStream) to read response chunks without buffering the entire response. */
-export default function fetchRequest(options: TransportOptions): Transport {
-  options.debug && debug("fetchRequest", options);
-  return new Fetch(options);
+export default function fetchRequest(options: TransportOptions, fetchOptions: FetchOptions = {}): Transport {
```

Comment:
Do we need both of these functions? It seems like a bit of duplication to me. Can you quantify the benefit of having both? Personally I think maybe just this one would suffice.

Reply:
The `fetchRequest` is preserved (but extended with additional optional argument) to keep the existing API intact.

Usage with the API before this PR (may not precise code):

```typescript
grpc.invoke({
  // ...
  transport: fetchRequest
})
```

Notice `fetchRequest` is passed as value, and is not called directly. This means there's no way to pass that additional argument - as it's called internally by the library.

With the new `fetchRequestWithOptions` we'd use it like this:

```typescript
grpc.invoke({
  // ...
  transport: fetchRequestWithOptions({ credentials: "include" })
})
```

This effectively creates a carried version of the `fetchRequest` with the options we need to pass, allowing the desired usecase.

I'm not against changing any of that, I just wanted this PR to not break the existing APIs.

Reply:
So `fetchRequestWithOptions` is required to access this new API? Of course, not breaking existing API is essential. I'm just showing my lack of TypeScript experience, thank you for clarifying.